crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1112,7 +1154,6 @@ impl<'db> Type<'db> {
             | (Type::Instance(InstanceType { class }), Type::StringLiteral(..)) => {
                 !matches!(class.known(db), Some(KnownClass::Str | KnownClass::Object))
             }
-            (Type::StringLiteral(..), _) | (_, Type::StringLiteral(..)) => true,
```

Comment:
Am I understanding right that this line (along with the similar ones removed below for `BytesLiteral` and `SliceLiteral`) were already redundant before this PR, because they were caught by the big union for single-valued types above?

Reply:
Exactly. And the compiler can now verify that this is the case, which led to me getting lots of warnings about unreachable branches locally.